File: document_store/src/main/kotlin/it/polito/document_store/dtos/DocumentDTO.kt
package it.polito.document_store.dtos

import it.polito.document_store.utils.DocumentCategory
import java.time.LocalDateTime

data class DocumentDTO(
    val name: String,
    val size: Long,
    val contentType: String,
    val category: DocumentCategory,
    /**
     * This value assumes different meaning based on the value of `category` field.
     * If the value of `category` field is equal to `Curriculum` the id is related to
     * a `professional` entity on the `crm microservice`.
     * */
    val id: Long? = null,
    val data: String
)

File: crm_ui_client/src/api/document_store/dto/DocumentDTO.ts
import {DocumentCategory} from "./DocumentMetadata";

export class DocumentDTO {
    name: string
    size: number
    contentType: string
    category: DocumentCategory
    id: number | null
    data: string

    constructor(
        name: string,
        size: number,
        contentType: string,
        category: DocumentCategory,
        id: number | null,
        data: string
    ) {
        this.name = name
        this.size = size
        this.contentType = contentType
        this.category = category
        this.id = id
        this.data = data
    }
}